specifyweb/frontend/js_src/lib/components/PickLists/TreeLevelPickList.tsx

```diff
@@ -142,11 +142,14 @@ export function TreeLevelComboBox(props: DefaultComboBoxProps): JSX.Element {
         if (destructorCalled) return undefined;
         const definitionItem = resource?.get('definitionItem');
 
+        const invalidDefinitionItem =
+          typeof definitionItem !== 'string' ||
+          !(items?.map(({ value }) => value).includes(definitionItem) ?? true);
+
         if (
           (items !== undefined ||
             typeof resource?.get('parent') !== 'string') &&
-          (typeof definitionItem !== 'string' ||
-            !items?.map(({ value }) => value).includes(definitionItem))
+          invalidDefinitionItem
```

Comment:
Maybe things can be simplified if we do not have a resource.onchange here. Or does that create more bugs? The items would be set by the previous effect so this second effect can just have items as part of the dependency.

Reply:
Fixed in https://github.com/specify/specify7/pull/4811/commits/e02e4eb4bbac53b161c4d58ecfcebc9ce7f23efd